vng_api_common/notifications/handlers.py

```diff
@@ -8,4 +17,44 @@ def handle(self, message: dict) -> None:
         logger.info("Received notification %r", message)
 
 
+class AuthHandler:
+    def handle_auth(self, message: dict) -> None:
+
+        # uuids in AC DB are synchronized with uuids in local DB
+        uuid = get_identifier_from_path(message['resource_url'])
+        
+        if message['actie'] == 'delete':
+            Applicatie.objects.get(uuid=uuid).delete()
+            return
+
+        # get info
+        applicatie_data = self._request_auth(message['resource_url'])
+        applicatie_data['uuid'] = uuid
+
+        # update models
+        try:
+            applicatie = Applicatie.objects.get(uuid=uuid)
+        except Applicatie.DoesNotExist:
+            applicatie_serializer = ApplicatieUuidSerializer(data=applicatie_data)
+        else:
+            applicatie_serializer = ApplicatieUuidSerializer(applicatie, data=applicatie_data)
+        applicatie_serializer.is_valid()
+        applicatie_serializer.save()
+        
+    def _request_auth(self, path: str) -> dict:
+        Client = import_string(settings.ZDS_CLIENT_CLASS)
+        client = Client.from_url(path)
+        response = client.retrieve('applicatie', path)
+        return underscoreize(response)
+
+    def handle(self, message: dict) -> None:
+        if message['kanaal'] == 'autorisaties':
+            self.handle_auth(message)
+
+        else:
```

Comment:
not liking this if-branch tbh

I think it'd make more sense to have a handler that routes based on `kanaal` to other handlers

something like:

```python
import logging


class LoggingHandler:

    def handle(self, message: dict) -> None:
        logger = logging.getLogger('notifications')
        logger.info("Received notification %r", message)


class AuthHandler:
    # everything else
    pass


class RoutingHandler:

    def __init__(self, config: dict, default=None):
        self.config = config
        self.default = default

    def handle(self, message: dict):
        handler = self.config.get(message['kanaal'])
        if handler is not None:
            handler.handle(message)
        elif self.default:
            self.default.handle(message)


log = LoggingHandler()
auth = AuthHandler()

default = RoutingHandler({'autorisaties': auth}, default=log)
```